aws/table_aws_backup_vault.go

```diff
@@ -264,6 +274,44 @@ func getAwsBackupVaultNotification(ctx context.Context, d *plugin.QueryData, h *
 	return op, nil
 }
 
+func getAwsBackupVaultTags(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	// Create Session
+	svc, err := BackupClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_backup_vault.getAwsBackupVaultTags", "connection_error", err)
+		return nil, err
+	}
+	if svc == nil {
+		// Unsupported region, return no data
+		return nil, nil
+	}
+
+	if svc == nil {
+		// Unsupported region, return no data
+		return nil, nil
+	}
+
+	arn := vaultArn(h.Item)
+
+	params := &backup.ListTagsInput{
+		ResourceArn: aws.String(arn),
+	}
+
+	op, err := svc.ListTags(ctx, params)
+	if err != nil {
+
+		var ae smithy.APIError
+		if errors.As(err, &ae) {
+			if ae.ErrorCode() == "ResourceNotFoundException" {
+				return &backup.GetBackupVaultNotificationsOutput{}, nil
+			}
+		}
```

Comment:
Beware of the caveat that when you try to list tags for a non-existent vault you get a different error instead:
```
 aws backup list-tags --resource-arn arn:aws:backup:us-east-2:AWS_ACCOUNT_ID:backup-vault:NonExistent

An error occurred (AccessDeniedException) when calling the ListTags operation: Insufficient privileges to perform this action.
```

This is actually a frequent pattern in AWS (although not a consistent ones). AWS shows returns an `AccessDenied` error, so that if the caller doesn't have permissions to list tags for some vaults it doesn't even know if the vault they tried to list tags for exists or not.

However, in general, I'd like to return to the original `X` question. Should there even be this special handling for such kind of error here? If the vault doesn't exist or the user can't list its tags for any other reason, then steampipe should exit with an error, right?

Reply:
Typically, we manage `Not Found Error Codes` for the APIs within the tables. However, it's worth noting that we deliberately haven't addressed 'AccessDeniedException' when making API calls.

For those who prefer receiving an empty row rather than encountering 'AccessDeniedException,' Steampipe offers the option to achieve this by configuring the [ignore_error_codes](https://hub.steampipe.io/plugins/turbot/aws#configuration) block in the configuration file located at `~/.steampipe/config/aws.spc`. This configuration allows for a more customized handling of errors based on user preferences.